static_default_ports.go

```diff
@@ -0,0 +1,36 @@
+package rdns
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+)
+
+var (
+	DoQPort      string = "8853"
+	DohQuicPort  string = "1443"
+	DoTPort      string = "853"
+	DTLSPort     string = DoTPort
+	DoHPort      string = "443"
+	PlainDNSPort        = "53"
+)
+
+func AddressWithDefault(addr, defaultPort string) string {
```

Comment:
Sorry to keep bringing up this function. Since you're now dealing with different address formats, you can consider using https://pkg.go.dev/net/url#Parse to do the heavy lifting, it's going to be a lot more reliable. Consider doing it like this:

```go
if addr == "" {
  return addr
}
u, err := url.Parse(addr)
if err != nil {
  return addr
}
if u.Scheme == "http" || u.Scheme == "https" {
  return addr // do we want to set a port on those? Maybe not
}
if u.Port() == "" {
  u.Host := net.JoinHostPort(u.Host, defaultPort)
}

return u.String() // re-assemble the address, now with port

Reply:
I think we do want to set a port on `http` and `https` tho, because there are alt ports for both such as `8080` or `8443` which both require that scheme to be processed correctly

Reply:
Can certainly do that, it shouldn't hurt anything. I revised the code snippet from above to do that. I think it'll work, should add a test for it.
```go
if addr == "" {
  return addr
}
u, err := url.Parse(addr)
if err != nil {
  return addr
}
if u.Port() == "" {
  u.Host = net.JoinHostPort(u.Host, defaultPort)
}
if u.Scheme == "" { // no url, just host+port
  return u.Host
}
return u.String() // re-assemble the address, now with port
```

Reply:
I run into this error - 
```
zsh: exit 1     ./routedns example-config/well-known-wo-ports.toml
```
do to - 
```
[resolvers.cloudflare-doh-1-0-0-1-get]
address = "https://1.0.0.1/dns-query{?dns}" --- here
protocol = "doh"
doh = { method = "GET" }
```
I have no idea on a good way to fix it

Reply:
Did you use my code above? I can't repro this here. You can commit what you have and I can take a look

Reply:
I pretty much did this is what I used 
```
if addr == "" {
		return addr
	}
	u, _ := url.Parse(addr)
	if u.Port() == "" {
		u.Host = net.JoinHostPort(u.Host, defaultPort)
	}
	if u.Scheme == "" { // no url, just host+port
		return u.Host
	}
	return u.String() // re-assemble the address, now with port
``` 
I removed the error part after I figured out the `{}` was causing an error when parsing the url

Reply:
Turns out this had way more edge-cases than I expected. So I added a test and simplified the function a bit. It's https://github.com/folbricht/routedns/pull/202 which targets your branch. Feel free to take it as is onto your branch, or build on top of it. I may also have missed a few test cases, should be easy to add those now